sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaReadSchemaTransformProvider.java

```diff
@@ -139,6 +151,59 @@ Row getConfigurationRow() {
       }
     }
 
+    private PCollectionRowTuple expandRawWithKeyMetadata(
```

Comment:
There's a lot of duplicate code from `expand()` here :(

I'm a fan of making small code changes if possible. Does it make sense to consolidate this with the existing `expand()` code block?